src/ifcconvert/IfcConvert.cpp

```diff
@@ -429,6 +432,29 @@ int main(int argc, char** argv) {
 
     IfcGeom::Iterator<real_t> context_iterator(settings, input_filename);
 
+    if (!building_storey.empty()) {
+        bool found_storey = false;
+        IfcParse::IfcFile *ifc_file = context_iterator.getFile();
+        IfcEntityList::ptr storeys = ifc_file->entitiesByType("IfcBuildingStorey");
```

Comment:
Better use the templated version: `ifc_file->entitiesByType<IfcSchema::IfcBuildingStorey>()`, than you get a typed list back.